File: leet_BTREE_236_lowest_common_ancestor.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func lowestCommonAncestor(node, p, q *TreeNode) *TreeNode {
     if node == nil || node == p || node == q {
         return node
     }
     L := lowestCommonAncestor(node.Left, p, q)
     R := lowestCommonAncestor(node.Right, p, q)
     if (L == nil) {
         return R
     }
     if (R == nil) {
         return L
     }
     return node
}

File: leet_BTREE_236_lowest_common_ancestor.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {TreeNode}
 */
var lowestCommonAncestor = function(node, p, q) {
    if (node == null || node == p || node == q)
        return node
    let L = lowestCommonAncestor(node.left, p, q)
    let R = lowestCommonAncestor(node.right, p, q)
    if (!L)
        return R
    if (!R)
        return L
    return node
};
